packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -14,17 +13,56 @@ import 'package:flutter/scheduler.dart';
 ///
 /// ImageInfo objects are used by [ImageStream] objects to represent the
 /// actual data of the image once it has been obtained.
+///
+/// The receiver of an [ImageInfo] object must call [Image.dispose] on the
+/// [image]. To safely share the object with other clients, use the [clone]
+/// method.
 @immutable
 class ImageInfo {
   /// Creates an [ImageInfo] object for the given [image] and [scale].
   ///
-  /// Both the image and the scale must not be null.
+  /// Both the [image] and the [scale] must not be null.
   ///
-  /// The tag may be used to identify the source of this image.
+  /// The [debugLabel] may be used to identify the source of this image.
   const ImageInfo({ required this.image, this.scale = 1.0, this.debugLabel })
     : assert(image != null),
       assert(scale != null);
 
+  /// Creates an [ImageInfo] with a cloned [image].
+  ///
+  /// Once all outstanding references to the [image] are disposed, it is no
+  /// longer safe to access properties of it or attempt to draw it. Clones serve
+  /// to create new references to the underlying image data that can safely be
+  /// disposed without knowledge of whether some other reference holder will
+  /// still need access to the underlying image. Once a client disposes of its
+  /// own image reference, it can no longer access the image, but other clients
+  /// will be able to access their own references.
+  ///
+  /// This method must be used in cases where a client holding an [ImageInfo]
```

Comment:
concrete example of a Client is an Image widget? Or an image cache?

Reply:
Any method or class receiving this object.